plugins/chat/app/serializers/chat_message_serializer.rb

```diff
@@ -35,23 +35,23 @@ def excerpt
   end
 
   def reactions
-    reactions_hash = {}
     object
       .reactions
       .group_by(&:emoji)
-      .each do |emoji, reactions|
-        users = reactions[0..5].map(&:user).filter { |user| user.id != scope&.user&.id }[0..4]
-
+      .map do |emoji, reactions|
         next unless Emoji.exists?(emoji)
 
-        reactions_hash[emoji] = {
+        users = reactions.take(5).map(&:user).take(5)
```

Comment:
The second `take()` doesn't do anything, does it? It's not possible for `map()` to increase the size of an array?

Reply:
Yes I think you are right, the original code was already convulted :+1: